components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -117,6 +121,36 @@ void BraveReferralsService::Start() {
   if (initialized_)
     return;
 
+  base::PostTaskAndReplyWithResult(
+      FROM_HERE, {base::ThreadPool(), base::MayBlock()},
+      base::BindOnce(&HasPromoCodeFile, GetPromoCodeFileName()),
+      base::BindOnce(&BraveReferralsService::DoInitialization,
+                     weak_factory_.GetWeakPtr()));
+}
+
+void BraveReferralsService::ResetPrefs() {
+  pref_service_->ClearPref(kReferralCheckedForPromoCodeFile);
+  pref_service_->ClearPref(kReferralPromoCode);
+  pref_service_->ClearPref(kReferralDownloadID);
+  pref_service_->ClearPref(kReferralTimestamp);
+  pref_service_->ClearPref(kReferralAttemptTimestamp);
+  pref_service_->ClearPref(kReferralAttemptCount);
+  pref_service_->ClearPref(kReferralHeaders);
+#if defined(OS_ANDROID)
+  pref_service_->ClearPref(kReferralAndroidFirstRunTimestamp);
+#endif
+}
+
+void BraveReferralsService::DoInitialization(bool has_promo_code_file) {
+  DCHECK(!initialized_);
+
+  if (has_promo_code_file) {
+    // For first run, this is no-op.
+    // If it's not a first run, this means user installs another referrals.
+    // Let's start new referrals after clearing existing prefs.
+    ResetPrefs();
```

Comment:
Just curious: how does a user go about installing a new referral to trigger this code? Are they running a new installer? It seems not, since otherwise there would be no preferences to reset.

One thing about the referral code that I remember (at least on Windows) is that the Omaha installer writes the referral code to the registry. Just wasn't sure if that's something we need to take into consideration here.

Other than that, this approach seems fine to me.

Reply:
Yes, user will install super referrals by using new installers that super-referrer code appended to its installer filename.
Then, new promocode file will be copied to existing user dir.

I'll also check how windows omaha installer works.

Reply:
I wrote about how promoCode file is generated in user dir in description.
and I checked that whenever promo-installer(or pkg installer) runs, it generates promoCode file.

Reply:
So, I think this PR will works well for our purpose on macOS and Windows.
Let's wait @aekeus 's opinion about backend.